store/src/commonMain/kotlin/org/mobilenativefoundation/store/store5/StoreBuilder.kt

```diff
@@ -54,22 +53,61 @@ interface StoreBuilder<Key : Any, Output : Any> {
          *
          * @param fetcher a [Fetcher] flow of network records.
          */
-        @OptIn(ExperimentalTime::class)
-        fun <Key : Any, Output : Any> from(
-            fetcher: Fetcher<Key, Output>
-        ): StoreBuilder<Key, Output> = RealStoreBuilder(fetcher)
+        fun <Key : Any, CommonRepresentation : Any> from(
+            fetcher: Fetcher<Key, *>,
+        ): StoreBuilder<Key, CommonRepresentation> = storeBuilderFrom(fetcher = fetcher)
 
         /**
          * Creates a new [StoreBuilder] from a [Fetcher] and a [SourceOfTruth].
          *
          * @param fetcher a function for fetching a flow of network records.
          * @param sourceOfTruth a [SourceOfTruth] for the store.
          */
-        fun <Key : Any, Input : Any, Output : Any> from(
-            fetcher: Fetcher<Key, Input>,
-            sourceOfTruth: SourceOfTruth<Key, Input, Output>
-        ): StoreBuilder<Key, Output> = RealStoreBuilder(
+        fun <Key : Any, CommonRepresentation : Any> from(
+            fetcher: Fetcher<Key, *>,
+            sourceOfTruth: SourceOfTruth<Key, *>
+        ): StoreBuilder<Key, CommonRepresentation> = storeBuilderFrom(fetcher = fetcher, sourceOfTruth = sourceOfTruth)
+    }
+}
+
+interface MutableStoreBuilder<Key : Any, NetworkRepresentation : Any, CommonRepresentation : Any, SourceOfTruthRepresentation : Any, NetworkWriteResponse : Any> :
+    StoreBuilder<Key, CommonRepresentation> {
+    override fun build(): MutableStore<Key, CommonRepresentation, NetworkWriteResponse>
+
+    /**
+     * @see [StoreBuilder.scope]
+     */
+    override fun scope(scope: CoroutineScope): MutableStoreBuilder<Key, NetworkRepresentation, CommonRepresentation, SourceOfTruthRepresentation, NetworkWriteResponse>
+
+    /**
+     * @see [StoreBuilder.cachePolicy]
+     */
+    override fun cachePolicy(memoryPolicy: MemoryPolicy<Key, CommonRepresentation>?): MutableStoreBuilder<Key, NetworkRepresentation, CommonRepresentation, SourceOfTruthRepresentation, NetworkWriteResponse>
+
+    /**
+     * @see [StoreBuilder.disableCache]
+     */
+    override fun disableCache(): MutableStoreBuilder<Key, NetworkRepresentation, CommonRepresentation, SourceOfTruthRepresentation, NetworkWriteResponse>
+
+    fun converter(converter: StoreConverter<NetworkRepresentation, CommonRepresentation, SourceOfTruthRepresentation>): MutableStoreBuilder<Key, NetworkRepresentation, CommonRepresentation, SourceOfTruthRepresentation, NetworkWriteResponse>
+
+    companion object {
```

Comment:
I think we should have a factory that looks something like 

`store.AsMutableStore(updater, bookkeeper)`